restapi/src/main/java/io/stargate/web/docsapi/dao/DocumentDB.java

```diff
@@ -95,6 +95,10 @@ public DocumentDB(
     this.dataStore = dataStore;
     this.authToken = authToken;
     this.authorizationService = authorizationService;
+
+    if (!dataStore.supportsSAI() && !dataStore.supportsSecondaryIndex()) {
+      throw new IllegalStateException("Backend does not support any known index types.");
```

Comment:
It seems that in some places we just log that SAI is not supported (https://github.com/stargate/stargate/pull/528/files#diff-d36bfbe0c5f35c2d92be42df6186c221c2a80c49022c728aae5ccd217443be3aR226) but here we are throwing an error. Is it must have to have one of those indexes support?

Reply:
I believe so. When `DocumentDB` [creates indexes](https://github.com/stargate/stargate/pull/528/files#diff-d36bfbe0c5f35c2d92be42df6186c221c2a80c49022c728aae5ccd217443be3aR203-R206) it has only two options: SAI and "default". The "default" in this contex may sound a bit confusion, but really it mean "secondary" IIRC.

@EricBorczuk : WDYT?

Reply:
Right, it has to support at least one of them for docsapi. If SAI isn't supported would be good to just fallback to secondary.

Reply:
That's what Docs API does (fallback). This check is just to ensure we have at least one of the index kinds available to us.